src/protocol/boolean/bit_decomposition.rs

```diff
@@ -133,6 +281,38 @@ mod tests {
         result.reconstruct()
     }
 
+    async fn bit_addition<F, S, C>(
+        world: &TestWorld,
+        r: u32,
+        g_b: GBIterator,
+        q: F,
+        num_bits: u32,
+    ) -> Vec<F>
+    where
+        F: PrimeField + ExtendableField + Sized,
+        Standard: Distribution<F>,
+    {
+        let r_f_b = get_bits::<F>(r, num_bits);
+
+        let result = world
+            .semi_honest((q, r_f_b), |ctx, (q_p, r_b_d)| async move {
+                let ctx = ctx.set_total_records(1);
+                let r_b = r_b_d.to_vec();
+                compute_bit_addition(ctx, RecordId::from(0), r_b, g_b, &q_p)
```

Comment:
@akoshelev I'm getting an error here about moving `g_b`. Any idea how to fix it? Here's the full error:

```
error[E0507]: cannot move out of `g_b`, a captured variable in an `Fn` closure
   --> src/protocol/boolean/bit_decomposition.rs:298:58
    |
287 |           g_b: GBIterator,
    |           --- captured outer variable
...
298 |               .semi_honest((q, r_f_b), |ctx, (q_p, r_b_d)| async move {
    |  ______________________________________-------------------_^
    | |                                      |
    | |                                      captured by this `Fn` closure
299 | |                 let ctx = ctx.set_total_records(1);
300 | |                 let r_b = r_b_d.to_vec();
301 | |                 compute_bit_addition(ctx, RecordId::from(0), r_b, g_b, &q_p)
    | |                                                                   ---
    | |                                                                   |
    | |                                                                   variable moved due to use in generator
    | |                                                                   move occurs because `g_b` has type `GBIterator`, which does not implement the `Copy` trait
302 | |                     .await
303 | |                     .unwrap()
304 | |             })
    | |_____________^ `g_b` is moved here
```

Reply:
cc @martinthomson too 😀

Reply:
The callback identified is invoked 3 times.  You only have one copy of `g_b`.  Consider moving `g_b` into the tuple that you pass to `semi_honest()` (you might need to `collect()` it...)

Reply:
I tried moving it into that tuple, but `semi_honest` modifies the values passed to it (in this case, converting field elements into secret sharing.) When I included `g_b`, it complained about not having an implementation for that type of iterator, which makes sense, since I just made the iterator.